### PR TITLE
Add config feature 'initial_branch_name'

### PR DESCRIPTION
**Pull request by fnune**
This config feature allows users to specify an initial branch name that will be used as a `get_user_input` default when Neogit asks for a name for a new branch.

This is useful when working in teams with a branch naming convention. For example, teams may choose to prefix branches with a GitHub username. In that case, configuring Neogit with:

```
initial_branch_name = "myname/"
```

...will result in the popup being prepopulated with `myname/`, after which I can specify the branch name.

### CODE REVIEW

File: lua/neogit/popups/branch/actions.lua [hunk 1]
```diff
@@ -64,10 +68,7 @@ local function create_branch(popup, prompt, checkout)
     return
   end
 
-  local name = input.get_user_input("Create branch", {
-    strip_spaces = true,
-    default = popup.state.env.suggested_branch_name,
-  })
+  local name = get_branch_name_user_input("Create branch", popup.state.env.suggested_branch_name)
```
Comment (fnune) [inline lua/neogit/popups/branch/actions.lua:71]: I couldn't figure out how to test this behavior. How can I cause a situation where `popup.state.env.suggested_branch_name` is defined?

Comment (fnune) [reply lua/neogit/popups/branch/actions.lua:71]: I found that, but what do I need in the UI in order to get there?

**Comment by fnune** [comment]
Hi! @CKolkey do you think this can get reviewed/merged?

**Comment by CKolkey** [comment]
yea 😅 I'll try to take a look tomorrow. Sorry for the delay

**Comment by fnune** [comment]
Nothing to be sorry about :smile:

**Comment by fnune** [comment]
Btw @CKolkey have you considered adding the `#hacktoberfest` label to the repo? :) https://hacktoberfest.com/

**Comment by CKolkey** [comment]
> Btw @CKolkey have you considered adding the `hacktoberfest` label to the repo? :) [hacktoberfest.com](https://hacktoberfest.com/)

Sure, why not :)

**Pull request merged**